### PR TITLE
Geospatial utility class

### PR DESCRIPTION
**Pull request by fabrice-v**
**Summary:**

This PR adds a geospatial utilities class to the project in preparation for #137 and #155 . 
Third party libraries are encapsulated through the use of an interface.
It's modeled after what was done for time related utilities and includes name changes for consistency.

@barbeau I split this off in its own PR so it could be reviewed and merged independently of progress made on the two mentioned issues and provide a place in which to put any methods required by 137

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [ ] Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/geoutils/GeospatialUtilsImpl.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+     * @return a unique instance of {@code GeospatialUtilsImpl}
+     */
+    public static GeospatialUtilsImpl getInstance() {
+        if (GEO_UTILS == null) {
+            GEO_UTILS = new GeospatialUtilsImpl();
+        }
+        return GEO_UTILS;
+    }
+
+    /**
```
Comment (barbeau) [inline adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/geoutils/GeospatialUtilsImpl.java:43]: There is an alternate implementation of this in https://github.com/MobilityData/gtfs-validator/pull/302/files#diff-2a4deacfd60d7ef9315e71a0b66324beR53 that uses the `DistanceCalculator` class - https://locationtech.github.io/spatial4j/apidocs/com/spatial4j/core/distance/DistanceCalculator.html#distance(com.spatial4j.core.shape.Point,%20com.spatial4j.core.shape.Point).

Was the DistanceUtils class used for a reason here?

Comment (fabrice-v) [reply adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/geoutils/GeospatialUtilsImpl.java:43]: No other reason than not remembering the other PR and feeling like it is less code.

Comment (fabrice-v) [reply adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/geoutils/GeospatialUtilsImpl.java:43]: I'll ask @lionel-nj to backport his implementation in this PR.

Comment (lionel-nj) [reply adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/geoutils/GeospatialUtilsImpl.java:48]: @fabrice-v, see f936c7f7d6f190709b67687a4f9384faa7d08ae9.

**Pull request merged**